test/CMakeLists.txt

```diff
@@ -93,6 +93,7 @@ set(TEST_SOURCES
   ${CMAKE_CURRENT_SOURCE_DIR}/OperandCapabilities.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/Operand.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/OperandPattern.cpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/ParseNumber.cpp
```

Comment:
I'm hesitating of enlarging the already huge `UnitSPIRV` test binary. Can we have a separate test binary generated for this like what we do for optimizations?

Just need to add a new `CMakeLists.txt` in `test/utils/` to call `add_spvtools_unittest`.